kernel/objects/scheduling-context/objects/SchedulingContext.cc

```diff
@@ -67,11 +68,17 @@ namespace mythos {
         
         // add to the ready queue
         readyQueue.remove(ec); /// @todo do not need to remove if already on the queue, just do nothing then. This needs additional information in handle_t of LinkedList
-        readyQueue.push(ec);
+        priorityQueue.remove(ec); /// @todo do not need to remove if already on the queue, just do nothing then. This needs additional information in handle_t of LinkedList
+        if(ec->get()->hasPriority()){
+          priorityQueue.push(ec);
+          home->preempt(); // trigger reschedule to run priority ec 
+        }else{
+          readyQueue.push(ec);
 
-        // wake up the hardware thread if it has no execution context running
-	// or if if current ec got ready in case of race condition
-        if (current == nullptr || current == ec) home->preempt();
+          // wake up the hardware thread if it has no execution context running
+    // or if if current ec got ready in case of race condition
```

Comment:
If a running EC priority has been set to default from another hw thread (-> without interrupting it), this should also be enough to prevent the EC from running if there is another high priority EC waiting to run.

Reply:
yes, but I don't want to handle this special case.